runhouse/main.py

```diff
@@ -669,27 +660,45 @@ def cluster_list():
         cluster_status = (
             den_cluster.get("status") if den_cluster.get("status") else "unknown"
         )
-        if cluster_status == "running":
-            running_clusters.append(
-                {
-                    "Name": cluster_name,
-                    "Cluster Type": cluster_type,
-                    "Status": cluster_status,
-                }
-            )
-        else:
-            not_running_clusters.append(
-                {
-                    "Name": cluster_name,
-                    "Cluster Type": cluster_type,
-                    "Status": cluster_status,
-                }
-            )
+
+        # currently relying on status pings to den as a sign of cluster activity.
+        # The split is required to remove milliseconds and the offset (according to UTC) from the timestamp.
+        # (status_last_checked is in the following format: YYYY-MM-DD HH:MM:SS.ssssss±HH:MM)
+
+        last_active_at = den_cluster.get("status_last_checked")
+        last_active_at = (
+            datetime.datetime.fromisoformat(last_active_at.split(".")[0])
+            if isinstance(last_active_at, str)
+            else None
+        )
+        last_active_at = last_active_at.replace(tzinfo=datetime.timezone.utc)
+
+        if cluster_status == "running" and not last_active_at:
+            # For BC, in case there are clusters that were saved and created before we introduced sending cluster status to den.
+            cluster_status = "unknown"
+
+        cluster_info = {
+            "Name": cluster_name,
+            "Cluster Type": cluster_type,
+            "Status": cluster_status,
+            "Last Active (UTC)": last_active_at,
```

Comment:
do you have a screenshot you could share for this PR similar to the previous one showing how it would look in the CLI?

Reply:
![image](https://github.com/user-attachments/assets/cd12c517-2b32-48ec-a101-522ad8c333b5)